device/transport/amqp/src/amqp_twin_client.ts

```diff
@@ -221,13 +214,6 @@ export class AmqpTwinClient extends EventEmitter {
             this._fsm.deferUntilTransition('detached');
             this._fsm.transition('detaching');
           },
-          handleLinkDetach: (detachObject) => {
```

Comment:
>handleLinkDetach: (detachObject) => { [](start = 10, length = 37)

While I'm trying to recall my thinking here,  The twin client upon receiving a detach, for ONE of the two links that created to perform twin communication, will try to close the other.  As written I don't believe it has a way to determine what the other is.  (If original detach occurs on the upstream link, I don't recall whether the detach on handler can determine that it was the upstream.  Consequently, it tries to kill both.  Are you saying the base amqp client will automagically detach both of these links.?

Reply:
yes

---
In reply to: [167287836](https://github.com/Azure/azure-iot-sdk-node/pull/228#discussion_r167287836) [](ancestors = 167287836)

Reply:
and for the record all the e2e tests (including disconnection and detach of the links) pass with this modification.

---
In reply to: [167311755](https://github.com/Azure/azure-iot-sdk-node/pull/228#discussion_r167311755) [](ancestors = 167311755,167287836)